File: propertyUtils/PropertyUtils.java
package propertyUtils;

public class PropertyUtils {

    /**
     * Return Zim Common Utils prop file
     *
     * @since 09.05.2021
     * @return Property entity of the zim.global.properties file
     */
    public static Property getGlobalPropertyEntity() {
        return new Property();
    }

    /**
     * Return a value from the key based on zim global properties file
     * @param key  the key in the properties file
     * @since 11.05.2021
     * @return Property entity of the zim.global.properties file
     */
    public static String getGlobalProperty(String key) {
        return new Property().getProperty(key);
    }
}

File: common_utils/propertyUtils/property_utils.py
"""Utility helpers mirroring the Java ``PropertyUtils`` class."""

from __future__ import annotations

from typing import Optional

from .property import Property


def get_global_property_entity(path_to_property: Optional[str] = None) -> Property:
    """Return a :class:`Property` instance for the global properties file."""

    return Property(path_to_property)


def get_global_property(key: str) -> Optional[str]:
    """Fetch a value from the global properties file by ``key``."""

    return Property().get_property(key)


__all__ = ["get_global_property", "get_global_property_entity", "Property"]
